Sources/TuistCore/GraphTraverser/GraphTraversing.swift

```diff
@@ -107,6 +107,12 @@ public protocol GraphTraversing {
     ///   - name: Target name.
     func appExtensionDependencies(path: AbsolutePath, name: String) -> Set<GraphTarget>
 
+    /// Given a project directory and a target name, it returns all the dependencies that are extensions with ther condition
+    /// - Parameters:
+    ///   - path: Path to the directory that contains the project.
+    ///   - name: Target name.
+    func appExtensionDependenciesWithConditions(path: AbsolutePath, name: String) -> [(GraphTarget, PlatformCondition?)]
```

Comment:
This can replace `appExtensionDependencies`, I don't think we'd need both.

Reply:
I agree, but I was going to take a pass at refactoring all of these approaches to a single type and I can handle that in a latter PR if @alexanderwe wants to just move this one forward.